probe_scraper/runner.py

```diff
@@ -310,15 +311,26 @@ def sync_output_and_cache_dirs(output_bucket, cache_bucket, out_dir, cache_dir,
         sys.exit(1)
     else:
         print("Syncing output dir {}/ with s3://{}/".format(out_dir, output_bucket))
-        # The cloudfront dist will automatically gzip objects. Upload to s3.
-        sync_cft_cmd = (
-            f"aws s3 sync {out_dir}/ s3://{output_bucket}/ "
-            f"--delete "
-            f"--content-type 'application/json' "
-            f"--cache-control 'max-age=28800' "
-            f"--acl public-read"
-        )
-        os.system(sync_cft_cmd)
+
+        # cloudfront is supposed to automatically gzip objects, but it won't do that
+        # if the object size is > 10 megabytes (https://webmasters.stackexchange.com/a/111734)
+        # which our files sometimes are. to work around this, we'll regzip the contents into a
+        # temporary directory, and upload that with a special content encoding
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            for filename in os.listdir(out_dir):
```

Comment:
`os.listdir` will include both files and directories which will be problematic later. I think `os.walk` might be better suited here which will split out directories and files.

Reply:
this is a bit of a pain but you are probably right... fixing

Reply:
Done, it is a little complex so would be good to test again before deploying